lib/nat/service.go

```diff
@@ -258,10 +258,12 @@ func (s *Service) verifyExistingLocked(ctx context.Context, mapping *Mapping, na
 
 			l.Debugf("Renewed %s -> %s mapping on %s", mapping, address, id)
 
-			if !addr.Equal(address) {
-				mapping.removeAddressLocked(id)
-				mapping.setAddressLocked(id, addr)
-				change = true
+			for _, addr := range addrs {
+				if !addr.Equal(address) {
+					mapping.removeAddressLocked(id)
+					mapping.setAddressLocked(id, addr)
+					change = true
+				}
```

Comment:
This looks dodgy in the case you get multiple addresses, as it would replace it with the last one, etc.

Reply:
Yes, that's something that I still have to look into. There's also some code above that does a check involving the local IPv4 address

Reply:
I've changed `Mapping` now so it has a list of IP addresses per device and replaced this code to handle that with about the same semantics as before (report if the IP address(es) mapped on a gateway have changed). I've also made it skip the aforementioned check for IPv6, since I don't really think it makes sense there.